api/src/main/java/io/cloudevents/extensions/DatarefExtension.java

```diff
@@ -0,0 +1,86 @@
+/*
+ * Copyright 2018-Present The CloudEvents Authors
+ * <p>
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * <p>
+ * http://www.apache.org/licenses/LICENSE-2.0
+ * <p>
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+package io.cloudevents.extensions;
+
+import io.cloudevents.CloudEvent;
+import io.cloudevents.Extension;
+
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.Map;
+
+public final class DatarefExtension implements Extension {
+
+    public static final String DATAREF = "dataref";
+
+    private String dataref;
```

Comment:
Since dataref from the spec is a [`uri-reference`](https://github.com/cloudevents/spec/blob/master/extensions/dataref.md#dataref), i think we should use as type `URI`